src/utils/backend/person/teacher.ts

```diff
@@ -57,3 +63,55 @@ export async function createTeacher(
 
   return { data: createdTeacher, error: null };
 }
+
+// https://supabase.com/docs/reference/javascript/select
+
+export async function getTeacherList(classID: number): Promise<Teacher[]> {
+  // Get the teachers of all subjectRooms where class matches
+  const { data: roomSubjects, error: roomSubjectsError } = await supabase
+    .from<RoomSubjectDB>("room_subjects")
+    .select("teacher")
+    .match({ class: classID });
+  if (roomSubjectsError || !roomSubjects) {
+    console.error(roomSubjectsError);
+    return [];
+  }
+
+  // Map array of teacher IDs into array of teachers (fetch teacher in map)
+  const selected_teachers: (TeacherDB | null)[] = await Promise.all(
+    // 2. Flatten the arrays into an array of teacher IDs
+    roomSubjects
+      .map((roomSubject) => roomSubject.teacher)
+      .flat()
+      // remove duplicates
+      .filter((id, index, self) => self.indexOf(id) === index)
+      // 3. Fetch teacher data for each teacher
+      .map(async (teacher_id) => {
+        // TODO: Get teacher from ID (select only id, name, contacts, subject group)
```

Comment:
I'm not sure I understand these comments… (where's 1?) also remove TODOs if you're done with them.